internal/opnsense-unbound/client.go

```diff
@@ -8,26 +8,37 @@ import (
 	"fmt"
 	"io"
 	"net/http"
+	"net/url"
+	"path"
+	"strings"
 
 	log "github.com/sirupsen/logrus"
 	"sigs.k8s.io/external-dns/endpoint"
 )
 
+const emptyJSONObject = "{}"
+
 // httpClient is the DNS provider client.
 type httpClient struct {
 	*Config
 	*http.Client
+	baseURL *url.URL
 }
 
 const (
-	opnsenseUnboundServicePath  = "%s/api/unbound/service/%s"
-	opnsenseUnboundSettingsPath = "%s/api/unbound/settings/%s"
+	opnsenseUnboundServicePath  = "service"
+	opnsenseUnboundSettingsPath = "settings"
 	// Hacky, but nice to have the delete as an explicit constant since it's destructive
-	opnsenseUnboundSettingsPathDelete = "%s/api/unbound/settings/delHostOverride/%s"
```

Comment:
Does it make sense to change `"settings/delHostOverride"` to `opnsenseUnboundSettingsPath + "/delHostOverride"` also, or is that going overboard with DRY?

Reply:
It's hard to tell. Very few of these are used more than once, which is why I asked in another comment whether the constants should be kept at all. If they were to be kept, IMO it would make more sense to just have explicit lists rather than lots of string additions.

```
"settings"
"settings/delHostOverride"
```

Reply:
breaking it out into a library would also ensure it being the most DRY and the code here doesn't need any of that as well, but everything in due time